### PR TITLE
Make task selection report problems 

### PR DESCRIPTION
**Pull request by hegyibalint**
Task selection problems were not always reported as Problems.
This PR fixes this issue, fixes the categorization problems, and uses the new contextual labels to provide a custom message per task matching failure.

### CODE REVIEW

File: subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/TaskErrorExecutionIntegrationTest.groovy [hunk 1]
```diff
@@ -182,6 +183,9 @@ class TaskErrorExecutionIntegrationTest extends AbstractIntegrationSpec implemen
         fails "someTest"
 
         then:
+        verifyAll(receivedProblem(0)) {
+            fqid == 'task-selection:no-matches'
+        }
```
Comment (reinsch82) [inline subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/TaskErrorExecutionIntegrationTest.groovy:188]: 🤔 
Should we also test other parts of the problem?
like label?

**Review by reinsch82 — APPROVED** [review]
LGTM.
One question though.

File: subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/TaskErrorExecutionIntegrationTest.groovy (hunk 1 above)
Comment (hegyibalint) [reply subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/TaskErrorExecutionIntegrationTest.groovy:188]: The contextual label for sure

**Review by reinsch82 — APPROVED** [review]
LGTM

**Pull request merged**